library.cpp

```diff
@@ -38,33 +38,66 @@ boost::mutex mtx_;
 // To indicate if the server is running or not
 bool Is_Connected = false;
 
-
-// Convert a wide Unicode string to an UTF8 string
-std::string utf8_encode(const std::wstring &wstr)
+std::string utf8_encode(std::wstring const& wstr)
 {
-    if (wstr.empty()) return std::string();
+    if (wstr.empty()) {
+        return {};
+    }
+#if _WIN32 || _WIN64
     int size_needed = WideCharToMultiByte(CP_UTF8, 0, &wstr[0], (int)wstr.size(), NULL, 0, NULL, NULL);
     std::string strTo(size_needed, 0);
     WideCharToMultiByte(CP_UTF8, 0, &wstr[0], (int)wstr.size(), &strTo[0], size_needed, NULL, NULL);
     return strTo;
+#else
+    std::mbstate_t state{};
+
+    auto in         = wstr.c_str();
+    int size_needed = std::wcsrtombs(nullptr, &in, wstr.size(), &state);
+
+    std::string strTo(1 + size_needed, 0);
+    size_t n = std::wcsrtombs(strTo.data(), &in, wstr.size(), &state);
```

Comment:
note this was fixed in the subsequent commit. I should probably have cleaned up for a PR, then again, the PR is mainly the most convenient way to share the review, not so much intended as verbatim contribution